### ISSUE TITLE
gradCheck test

### ISSUE THREAD

**Issue by ronanmoynihan** [twitter/torch-autograd#52]
Hi,

I have a model which uses a normal Torch NN with an Autograd custom criterion. The gradCheck call returns true but I think I may be using it incorrectly.

The following file contains the test,
https://github.com/ronanmoynihan/autograd-criterion/blob/master/gradCheck.lua

If I comment out the following lines (27-30)  then the gradCheck fails, even though the model is not referenced anywhere else in the code.

local model = nn.Sequential()           
model:add(nn.Linear(d, numhid1)) 
model:add(nn.Sigmoid())
model:add(nn.Linear(numhid1, n_outputs))

**Comment by luketwitter** [comment]
Hi @ronanmoynihan ,

The following is not safe:
```
if type(inputs.T)~='table' then
```
Autograd wraps all your tensor input parameters, to allow them to be tracked throughout the function. So while you can apply the usual torch functions / operators to them, you shouldn't check the types. 

It seems like you were trying to work around the fact that assignment in the form of:

delta[torch.gt(delta,0)] =

Is not currently a supported operation. Let me talk to the team and see what can be done.

Luke

**Comment by luketwitter** [comment]
Also, using the : operator is not valid. You'll have to rewrite:
```
inputs.x:size(1)
```
to
```
torch.size(inputs.x, 1)
```
etc

**Comment by ronanmoynihan** [comment]
Hi @luketwitter,
OK, yeah I was thinking those commands (.gt & .lt) were not supported when I first started using it but managed to get a model training somewhat as I expected which gave similar predictions to a model that wasn't using Autograd.

This is a simple test on the Boston housing data and similar to the gradCheck linked above using a normal Torch NN with an Autograd criterion.

https://github.com/ronanmoynihan/autograd-criterion/blob/master/main.lua

```
th main.lua -under
```

**Comment by alexbw** [comment]
@ronanmoynihan Are you still having issues or can I close?

**Comment by ronanmoynihan** [comment]
Hi @alexbw you can close it.
Thanks.
